### PR TITLE
Add secrets and restart policy

### PR DESCRIPTION
**Pull request by davigar15**

### CODE REVIEW

File: opslib/osm/pod.py [hunk 1]
```diff
@@ -151,14 +154,17 @@ def volume_config(self):
...
-            }
-        )
+    def add_volume_config(self, name, mount_path, files, secret_name: str = None):
+        volume_config = {
+            "name": name,
+            "mountPath": mount_path,
+        }
+        if secret_name:
+            volume_config["secret"] = {"name": secret_name, "files": files}
+
```
Comment (ed1000) [inline opslib/osm/pod.py]: remove line

File: opslib/osm/pod.py [hunk 2]
```diff
@@ -273,6 +279,15 @@ def add_env(self, key: str, value: str):
     def add_envs(self, envs: dict):
         self._envs = {**self._envs, **envs}
 
+    def add_secret_env(self, secret_name: str, key: str, secret_key: str):
```
Comment (ed1000) [inline opslib/osm/pod.py]: why do you have two ways of doing the same thing? This is confusing

File: opslib/osm/pod.py [hunk 3]
```diff
@@ -294,6 +309,69 @@ def build(self):
...
+        """
+        Return a hash of the Pod Spec taking into account the current policy.
+        This means that for the hash, only the data included in the policy will be added
+
+        :param: pod_spec: Pod spec
+        """
+        spec = pod_spec.copy()
+        policy_spec = {"kubernetesResources": {}}
+
+        for resource_name, resource_names in self.default_policy[
```
Comment (ed1000) [inline opslib/osm/pod.py]: Confusing naming of variables

Comment (ed1000) [reply opslib/osm/pod.py]: resource_type, resource_names would be better, no?

File: opslib/osm/utils.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    """Get a hash from a dictionary"""
+    dict_str = json.dumps(dict, sort_keys=True)
+    result = hashlib.md5(dict_str.encode())
+    return result.hexdigest()
+
+
+def find_in_list_with_key(list: List[Dict[str, Any]], key: str, value: str) -> Any:
+    for item in list:
+        if key in item and item[key] == value:
+            return item
```
Comment (ed1000) [inline opslib/osm/utils.py]: Default return should be added for easier understanding that if the item is not found a None should be returned

File: opslib/osm/pod.py (hunk 1 above)
Comment (davigar15) [reply opslib/osm/pod.py]: done

File: opslib/osm/pod.py (hunk 2 above)
Comment (davigar15) [reply opslib/osm/pod.py]: done

File: opslib/osm/pod.py (hunk 3 above)
Comment (davigar15) [reply opslib/osm/pod.py]: done

File: opslib/osm/utils.py (hunk 4 above)
Comment (davigar15) [reply opslib/osm/utils.py]: done. Good idea btw

**Review by ed1000 — APPROVED** [review]

**Pull request merged**